lib/HunkView.ml

```diff
@@ -1,7 +1,86 @@
 open Nottui
 module W = Nottui_widgets
 
-(* Implementation of Single View Mode *)
+(* Common Functions *)
+
+type line =
+  | Change of string * WordDiff.word_diff list
+  | Common of string
+  | Empty
+
+let line_to_string = function Change (s, _) -> s | Common s -> s | Empty -> ""
+
+let construct_hunk mine their =
```

Comment:
This can be refactored into the word-diff module and needs some changes to process and construct the hunk properly as in the other comment.